web/beacon-app/src/features/topics/components/RevokeAPIKeyModal.tsx

```diff
@@ -55,6 +50,22 @@ const RevokeAPIKeyModal = ({ onOpen, onClose }: RevokeAPIKeyModalProps) => {
     }
   }, [wasKeyDeleted, onClose, reset, handleCheckboxChange]);
 
+  // handle error
+  useEffect(() => {
+    if (error && error.response.status === 401) {
```

Comment:
If a user's role is changed to `Observer` while they're logged in, they're still able to delete an API key until they log out. The role updates as expected, but they still have the ability to delete a key until they log out. This likely won't happen often, but this appears to be an edge case that will have to be handled. I just wanted to make a note so that this isn't forgotten.